postgrest/postgrest-postgis-spatial-api.md

```diff
@@ -0,0 +1,403 @@
+            3857
+          )
+        )/1000000 AS numeric
+      ),2
+    ) 
+
+$$ LANGUAGE SQL;
+```
+
+Let's give it a try.
+
+```sh
+curl -X POST \
+  http://localhost:3000/rpc/calc_area \
+  -H 'Content-Type: application/json' \
+  -H 'Prefer: params=single-object' \
+  -d '{"type":"FeatureCollection","features":[{"type":"Feature","properties":{},"geometry":{"type":"Polygon","coordinates":[[[-73.6962890625,41.02135510866602],[-73.8446044921875,41.253032440653186],[-74.168701171875,41.20758898181025],[-74.44335937499999,40.88444793903562],[-74.520263671875,40.60144147645398],[-74.2840576171875,40.53050177574321],[-73.8885498046875,40.63479884404164],[-72.83935546875,40.805493843894155],[-72.6800537109375,40.950862628132775],[-73.6962890625,41.02135510866602]]],"crs":{"type":"name","properties":{"name":"EPSG:4326"}}}}]}'
+  ```
+
+Our polygon covering part of New York is of size
+
+```
+[
+    {
+        "calc_area": 10723.31
+    }
+]
+```
+
+square kilometers.
+Easy.
+
+
+## Step 4 - Calculating the Intersection of 2 Polygons
+
+Last but not least let's implement one more function which will calculate the intersection of 2 polygons.
+The function by now should be self-explanatory. 
+We just have to make sure we are returning a `GeoJSON` and no longer a numeric value.
+
+
+```sh
+CREATE OR REPLACE FUNCTION api.calc_intersection(featurecollection json) RETURNS json AS $$ 
+
+      SELECT ST_AsGeoJSON(
+        ST_Intersection(
+          ST_GeomFromGeoJSON(
+            featurecollection->'features'->0->'geometry'
+          ),
+          ST_GeomFromGeoJSON(
+            featurecollection->'features'->1->'geometry'
+          )
+        )
+      )::json;
+    
+$$ LANGUAGE SQL;
+```
+
+We'll call this API with a feature collection comprising 2 polygons.
+
+
+```sh
+curl -X POST \
+  http://localhost:3000/rpc/calc_intersection \
+  -H 'Content-Type: application/json' \
+  -H 'Prefer: params=single-object' \
+  -d '{"type":"FeatureCollection","features":[{"type":"Feature","properties":{},"geometry":{"type":"Polygon","coordinates":[[[-73.6962890625,41.02135510866602],[-73.8446044921875,41.253032440653186],[-74.168701171875,41.20758898181025],[-74.44335937499999,40.88444793903562],[-74.520263671875,40.60144147645398],[-74.2840576171875,40.53050177574321],[-73.8885498046875,40.63479884404164],[-72.83935546875,40.805493843894155],[-72.6800537109375,40.950862628132775],[-73.6962890625,41.02135510866602]]],"crs":{"type":"name","properties":{"name":"EPSG:4326"}}}},{"type":"Feature","properties":{},"geometry":{"type":"Polygon","coordinates":[[[-73.6798095703125,41.56203190200195],[-74.3280029296875,41.75492216766298],[-74.34997558593749,41.74262728637672],[-74.7894287109375,41.21998578493921],[-74.29504394531249,40.87614141141369],[-73.4326171875,40.91766362458114],[-73.6798095703125,41.56203190200195]]],"crs":{"type":"name","properties":{"name":"EPSG:4326"}}}}]}'
+```
+
+Which in return will respond with the GeoJSON intersection.
+
+```sh
+[
```

Comment:
Can you get rid of the list type in the responses? E.g. this one would be way nicer, if it's only the GeoJSON without being nested.

The listed response is also unnecessary in all other examples.

Reply:
I tried but didn't find anything so far..